pkg/clusteragent/admission/mutate/agent_sidecar/agent_sidecar.go

```diff
@@ -129,12 +132,12 @@ func (w *Webhook) injectAgentSidecar(pod *corev1.Pod, _ string, _ dynamic.Interf
 	podUpdated := false
 
 	if !agentSidecarExists {
-		agentSidecarContainer := getDefaultSidecarTemplate(w.containerRegistry)
+		agentSidecarContainer := getDefaultSidecarTemplate(w.containerRegistry, w.datadogConfig)
```

Comment:
the `injectAgentSidecar()` is part of the hot execution path: each time we receive a request we run this function. That is why we should not call `config.Component` function in this function because retrieving config option value is costly.
For example we made this fix recently because we introduced a function that was using a config.Component function in a hot execution path. (see the fix: https://github.com/DataDog/datadog-agent/pull/29211)

So my suggestion is to make the `getDefaultSidecarTemplate()` function attached to the `Webhook` struct like `injectAgentSidecar()` and in the `NewWebhook()` call the the function `datadogConfig.GetString("admission_controller.agent_sidecar.image_name")` and store the value in `Webhook` struct. 

and do it for any other option value that we get with `datadogConfig.GetString(...)`.